packages/icon/src/LionIcon.js

```diff
@@ -85,16 +105,8 @@ export class LionIcon extends LitElement {
     this.__svg = svg;
     if (svg === undefined || svg === null) {
       this._renderSvg(nothing);
-    } else if (isPromise(svg)) {
```

Comment:
I think it tries to solve the same problem as the icon resolvers, it's better to have one solution than two. Dynamic imports for icons doesn't provide any added benefits, and is a worse DX I think.

Reply:
I agree with having one solution is a good DX, although it might make sense to also allow a single svg in apps that don't have many icons (or a few uncategorized).
We still have the api for synchronously loading an svg, so then I would say we keep async loading of single icons or remove that one as well.
Gonna discuss this internally with a few colleagues :)

Reply:
Ok, had some discussion and I think it's a good idea to do like your proposed: loading individual icons will become a perf bottleneck only if you have many of them and that's the moment you would create an iconset.